src/components/modal/ModalCreateBooks.tsx

```diff
@@ -0,0 +1,188 @@
+import AuthorService from '../../services/AuthorService'
+import placeholder from '../../assets/placeholder/placeholderForBook.png'
+
+
+interface Modal {
+  open: boolean,
+  onClose: () => void
+}
+
+
+const poratlDiv = document.getElementById('portal') as HTMLElement
+const ModalCreateBooks = ({ open , onClose } : Modal) => {
+
+  const [ showAddAuthor, setShowAddAuthor ] = useState(true)
+  const [ showAddButton, setShowAddButton ] = useState(true)
+  const [ authors, setAtuhors ] = useState([])
+  const [ fileImg, setFileImg ] = useState<Blob>(new Blob())
+  const [ coverImg, setCoverImg ] = useState('')
+  const [ booksData, setBooksData ] = useState<BookBody>({
+    Id: 0,
+    Title: '',
+    Description: '',
+    Isbn: '',
+    Quantity: 0,
+    Cover: fileImg,
+    PublishDate: '',
+    AuthorIds: []
+  })
+
+  const [ authorData, setAuthorData ] = useState<Author>({
+    Id: 0,
+    FirstName: '',
+    LastName: ''
+  })
+
+
+  const handleImgUpload = ({ currentTarget }: FormEvent<HTMLInputElement>) => {
+    if (currentTarget.files) {
+      const files = currentTarget.files
+      const reader = new FileReader()
+      if (files) {
+        reader.readAsDataURL(files[0])
+        setFileImg(files[0])
+        reader.onloadend = function () {
+          const base64data = reader.result
+          if (base64data) setCoverImg(base64data as string)
+        }
+      }
+    }
+  }
+  const hanlderCreateBooks = async (event: FormEvent<HTMLFormElement>) => {
+    event.preventDefault()
+    try {
+      const form = new FormData()
+      form.append('Title', booksData.Title)
+      form.append('Description', booksData.Description)
+      form.append('Isbn', booksData.Isbn)
+      form.append('Quantity', booksData.Quantity.toString())
+      form.append('Cover', fileImg)
+      form.append('PublishDate', booksData.PublishDate)
+      booksData.AuthorIds.forEach((author) => form.append('AuthorIds', author.Id.toString()))
+      await BookService.createBook(form)
+    } catch(error) {
+      if(axios.isAxiosError(error)) {
+        if(error.response?.status === 401) {
+          console.log('Not authenticated')
+        }
+      }
+    }
+  }
+
+  const handleCreateAuthor = async (event: FormEvent<HTMLFormElement>) => {
+    handlerNewAuthor()
+    event.preventDefault()
+    try {
+      const form = new FormData()
+      form.append('FirstName', authorData.FirstName)
+      form.append('LastName', authorData.LastName)
+      await AuthorService.createAuthor(form)
+    } catch(error) {
+      if(axios.isAxiosError(error)) {
+        if(error.response?.status === 401) {
+          console.log('Not authenticated')
+        }
+      }
+    }
+  }
+
+  const handlerNewAuthor = () => {
+    AuthorService.getAuthors().then((response) => {
+      setAtuhors(response)
+    })
+    setShowAddAuthor(true)
+    setShowAddButton(true)
+  }
+
+  const handlerCloseModal = () => {
+    onClose()
+  }
+
+  const handleFormAuthor = () => {
```

Comment:
Function name is not clear, what is this used for?

Reply:
I removed this handler, because i didn't need.